addons/l10n_sa/wizard/account_payment_register.py

```diff
@@ -0,0 +1,29 @@
+from odoo import api, fields, models
+
+
+class AccountPaymentRegister(models.TransientModel):
+    _inherit = 'account.payment.register'
+
+    l10n_sa_display_outstanding_account = fields.Boolean(compute='_compute_l10n_sa_display_outstanding_account')
+
+    @api.depends('partner_id', 'payment_type', 'journal_id')
+    def _compute_l10n_sa_display_outstanding_account(self):
+        for record in self:
+            moves = record.batches[0]['lines'].move_id
+            record.l10n_sa_display_outstanding_account = record.company_id.country_code == 'SA' and not \
+                                                         record.withholding_payment_account_id and \
+                                                         any(tax.tax_exigibility == 'on_payment' for tax in moves.line_ids.tax_ids)
+
+    def _compute_withholding_outstanding_account_id(self):
+        super()._compute_withholding_outstanding_account_id()
+        for record in self.filtered(lambda r: r.company_id.country_id.code == 'SA'
+                                    and not r.withholding_outstanding_account_id
+                                    and not r.withholding_payment_account_id):
+            payment_account = self.env.ref(f'account.{record.company_id.id}_account_journal_payment_credit_account_id', raise_if_not_found=False)
+            record.withholding_outstanding_account_id = payment_account
+
+    def _create_payment_vals_from_wizard(self, batch_result):
+        vals = super()._create_payment_vals_from_wizard(batch_result)
+        if self.company_id.country_code == 'SA' and self.withholding_outstanding_account_id and not vals.get('outstanding_account_id'):
+            vals['outstanding_account_id'] = self.withholding_outstanding_account_id.id
+        return vals
```

Comment:
Same here, why do you need to override this for SA? The payment withholding tax module already set this key when withholding taxes are in use.